### ISSUE TITLE
Trying to create nested page menu with datasource delegate 

### ISSUE THREAD

**Issue by cottageGrove** [tamanyan/SwiftPageMenu#34]
Hey tamanyan, 

Was just wondering if it was possible to create a nested page menu controller similar to Spotify's Music Podcasts Menu in your library.

![IMG_DF80B9305538-1](https://user-images.githubusercontent.com/17324193/74367908-ab2a3e80-4da0-11ea-9ca4-1f1c61f19966.jpeg)

```
    func viewControllers(forPageMenuController pageMenuController: PageMenuController) -> [UIViewController] {
        
        return self.items.map(ChildViewController.init)
        
    }
```

In the pageMenuControllerDataSource delegate method we can assign the View Controller. However It only takes a UIViewController and not a PageTabMenuViewController. Any kind of help would be appreciated. Thanks!

**Comment by tamanyan** [comment]
@cottageGrove I think you can create a nested page menu controller in terms of view level. However, I'm not sure how gestures works on the view. Have you passed `PageTabMenuViewController` as `ChildViewController` ?

**Comment by cottageGrove** [comment]
![IMG_9038A28F70DC-1](https://user-images.githubusercontent.com/17324193/74762999-e7e8b080-524c-11ea-8a5d-3f7f7c49344a.jpeg)


Sorry its more like this image above. Purple is the top level view and you can pass a subclass PageTabMenuVIewController to that tab whenever it is selected. Tried passing it as a`ChildViewController`but it can't cast down to a `PageTabMenuViewController`

**Comment by tamanyan** [comment]
@cottageGrove 

Is this what you expected?

<img width="584" alt="Screen Shot 2020-02-25 at 9 02 41 AM" src="https://user-images.githubusercontent.com/2387508/75208227-8f199a80-57ad-11ea-8afa-9addf7c770a0.png">

**Comment by tamanyan** [comment]
```swift
//
//  ChildPageTabMenuViewController.swift
//  PageMenuExample
//
//  Created by Tamanyan on 25/2/19.
//  Copyright © 2018 Tamanyan. All rights reserved.
//

import UIKit
import SwiftPageMenu

class ChildPageTabMenuViewController: PageMenuController {

    let items: [[String]]

    let titles: [String]

    init(items: [[String]], titles: [String], options: PageMenuOptions? = nil) {
        self.items = items
        self.titles = titles
        super.init(options: options)
    }

    required init?(coder aDecoder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }

    override func viewDidLoad() {
        super.viewDidLoad()

        self.edgesForExtendedLayout = []

        if options.layout == .layoutGuide && options.tabMenuPosition == .bottom {
            self.view.backgroundColor = Theme.mainColor
        } else {
            self.view.backgroundColor = .white
        }

        if self.options.tabMenuPosition == .custom {
            self.view.addSubview(self.tabMenuView)
            self.tabMenuView.translatesAutoresizingMaskIntoConstraints = false

            self.tabMenuView.heightAnchor.constraint(equalToConstant: self.options.menuItemSize.height).isActive = true
            self.tabMenuView.leftAnchor.constraint(equalTo: self.view.leftAnchor).isActive = true
            self.tabMenuView.rightAnchor.constraint(equalTo: self.view.rightAnchor).isActive = true
            if #available(iOS 11.0, *) {
                self.tabMenuView.bottomAnchor.constraint(equalTo: self.view.safeAreaLayoutGuide.bottomAnchor).isActive = true
            } else {
                self.tabMenuView.bottomAnchor.constraint(equalTo: self.view.bottomAnchor).isActive = true
            }
        }

        self.delegate = self
        self.dataSource = self
    }
}

extension ChildPageTabMenuViewController: PageMenuControllerDataSource {
    func viewControllers(forPageMenuController pageMenuController: PageMenuController) -> [UIViewController] {
        return self.items.map(ChildViewController.init)
    }

    func menuTitles(forPageMenuController pageMenuController: PageMenuController) -> [String] {
        return self.titles
    }

    func defaultPageIndex(forPageMenuController pageMenuController: PageMenuController) -> Int {
        return 0
    }
}

extension ChildPageTabMenuViewController: PageMenuControllerDelegate {
    func pageMenuController(_ pageMenuController: PageMenuController, didScrollToPageAtIndex index: Int, direction: PageMenuNavigationDirection) {
        // The page view controller will begin scrolling to a new page.
        print("didScrollToPageAtIndex index:\(index)")
    }

    func pageMenuController(_ pageMenuController: PageMenuController, willScrollToPageAtIndex index: Int, direction: PageMenuNavigationDirection) {
        // The page view controller scroll progress between pages.
        print("willScrollToPageAtIndex index:\(index)")
    }

    func pageMenuController(_ pageMenuController: PageMenuController, scrollingProgress progress: CGFloat, direction: PageMenuNavigationDirection) {
        // The page view controller did complete scroll to a new page.
        print("scrollingProgress progress: \(progress)")
    }

    func pageMenuController(_ pageMenuController: PageMenuController, didSelectMenuItem index: Int, direction: PageMenuNavigationDirection) {
        print("didSelectMenuItem index: \(index)")
    }
}
```

```swift
extension PageTabMenuViewController: PageMenuControllerDataSource {
    func viewControllers(forPageMenuController pageMenuController: PageMenuController) -> [UIViewController] {
       // Edit
        // return self.items.map(ChildViewController.init)
        return self.items.map { item in ChildPageTabMenuViewController(items: self.items, titles: self.titles) }
    }

    func menuTitles(forPageMenuController pageMenuController: PageMenuController) -> [String] {
        return self.titles
    }

    func defaultPageIndex(forPageMenuController pageMenuController: PageMenuController) -> Int {
        return 0
    }
}
```

**Comment by cottageGrove** [comment]
EXACTLY WHAT I NEEDED. Got it to work. I didn't know what the delegate was returning, but figured it out. I've posted the code below and some screen shots if you want to include this in the sample code. 

<img width="571" alt="Screen Shot 2020-02-25 at 3 56 53 PM" src="https://user-images.githubusercontent.com/17324193/75286939-729a5400-57e7-11ea-9fc2-0f6fd6baf426.png">

<img width="571" alt="Screen Shot 2020-02-25 at 3 57 15 PM" src="https://user-images.githubusercontent.com/17324193/75286962-7f1eac80-57e7-11ea-8a6d-ff76acdbacae.png">


```
extension PageTabMenuViewController : PageMenuControllerDataSource {
    func viewControllers(forPageMenuController pageMenuController: PageMenuController) -> [UIViewController] {
    
        let items1: [[String]] = [
            ["Trash", " Old Trash"],
            ["Even More Trash", "More Garbage"]
        ]
        
        let items2: [[String]] = [
            ["Eating Bagels", "Everything Bagel"],
            ["Rugalach", "More Bagels"]
        ]
        
        let headerTabTitles: [String] = [
            "GOAT",
            "MVP"
        ]
        
        let headerTabTitles2: [String] = [
            "BAGELS",
            "SANDWICHES"
            
        ]
        
        let items = [ChildPageTabMenuViewController(items: items1, titles: headerTabTitles), ChildPageTabMenuViewController(items: items2, titles: headerTabTitles2)]
        
        return items
        
    }

    func menuTitles(forPageMenuController pageMenuController: PageMenuController) -> [String] {
        return self.titles
    }

    func defaultPageIndex(forPageMenuController pageMenuController: PageMenuController) -> Int {
        return 0
    }
}
```
